### PR TITLE
Adds definitions for DrmAgentObject, ChannelConfig, and related objects

### PR DESCRIPTION
**Pull request by rathermarv**

### CODE REVIEW

**Review by joelKnapp — APPROVED** [review]

File: types/ChannelConfig.d.ts [hunk 1]
```diff
@@ -1,5 +1,48 @@
...
+        startFrequency: number;
+        endFrequency: number;
+        modulationModes: number;
+    }
+    export interface DVBTChannelScanParameters extends OIPF.ChannelScanParameters {
+        raster: number;
+        ofdm: string;
+        bandwidth: string;
+    }
+    export interface DVBSChannelScanParemeters extends OIPF.ChannelScanParameters {
```
Comment (BenSkidmoreDUK) [inline types/ChannelConfig.d.ts:16]: typo: Par'e'meters

File: types/FavouriteList.d.ts [hunk 2]
```diff
@@ -0,0 +1,3 @@
+declare namespace OIPF {
+    export type FavouriteList = OIPF.Collection<OIPF.Channel>;
```
Comment (BenSkidmoreDUK) [inline types/FavouriteList.d.ts:2]: I'm not entirely sure why ChannelList uses type Channel but FavouriteList uses type OIPF.Channel

**Review by BenSkidmoreDUK — CHANGES_REQUESTED** [review]
I've left two comments; one is a typo change request and the other is an uncertainty on my part regarding Channel object types.

File: types/FavouriteList.d.ts (hunk 2 above)
Comment (rathermarv) [reply types/FavouriteList.d.ts:2]: Good spot - just a lack of consistency on my part. Will add the namespace as that is the convention.

File: types/ChannelConfig.d.ts (hunk 1 above)
Comment (rathermarv) [reply types/ChannelConfig.d.ts]: Fixed now

**Review by BenSkidmoreDUK — APPROVED** [review]

**Pull request merged**